ecl/hql/hqlutil.cpp

```diff
@@ -10709,7 +10718,12 @@ IException * checkRegexSyntax(IHqlExpression * expr)
         {
             try
             {
-                if (isUnicodeType(expr->queryType()))
+                if (isUTF8Type(expr->queryType()))
+                {
+                    ICompiledStrRegExpr * compiled = rtlCreateCompiledU8StrRegExpr(rtlUtf8Size(value->getSize(), value->queryValue()), (const char *)value->queryValue(), false);
```

Comment:
getSize() is the size of the string, so no need to call rtlUtf8Size again.  I think possibly this should be a call to rtlUtf8Length().  Alternatively value->queryType()->getStringLen().

Reply:
Fixed.